src/reference/asciidoc/http.adoc

```diff
@@ -325,6 +326,30 @@ By default, `status-code-expression` is null meaning that the normal '200 OK' re
 ----
 
 The `<http:inbound-gateway>` resolves the 'status code' from the `http_statusCode` header of the reply Message.
+Starting with _version 4.2_, the default response status code when no reply is received within the `reply-timeout`
+is `500 Internal Server Error`.
+There are two ways to modify this behavior:
+
+- add a `reply-timeout-status-code-expression` - this has the same semantics as the `status-code-expression` on the
+inbound adapter.
+- Add an `error-channel` and return an appropriate message with an http status code header, such as...
+
+[source, xml]
+----
+<int:chain input-channel="errors">
+    <int:header-enricher>
+        <int:header name="http_statusCode" value="504" />
+    </int:header-enricher>
+    <int:transformer expression="payload.message" />
+</int:chain>
+----
+
+The payload of the `ErrorMessage` is a `MessageTimeoutException`; it must be transformed to something that can be
+converted by the gateway, such as a `String`; a good candidate is the exception's message property, which is the
```

Comment:
I don't understand the comment; if there's an error channel, the ErrorMessage contains this exception in the payload; if there's no error channel, the exception is thrown. It is caught in the http adapter and `getMessage()` is used as the payload.